Makefile

```diff
@@ -50,6 +50,18 @@ clean:
 test: generate operator-manifests
 	go test ./... -coverprofile cover.out
 
+# Run e2e-test
+e2e-test:
```

Comment:
Can you put `hack/prepare-e2e.sh` into `e2e.yaml` `steps` section and remove this target in makefile, so that we can directly issue command `e2e run -c test/e2e/e2e.yaml` instead of wrapping it in `make e2e-test`

Reply:
Take this as an example of how to install required tools in e2e, the idea is not to involve other tools (`make` in this case) as mush as possible

https://github.com/apache/skywalking/blob/b63008c61e1fcb1d69b25a8be7ce99f1d987a175/test/e2e-v2/cases/istio/als/e2e.yaml#L30-L36

Reply:
I have a small question here, I think we need to pull the required images firstly as `(install_images function) in hack/prepare-e2e.sh`, this is why `hack/prepare-e2e.sh` is executed before `e2e run -c test/e2e/e2e.yaml`. For other tools, I think we can add them in the `steps`  section.

Reply:
> I have a small question here, I think we need to pull the required images firstly as `(install_images function) in hack/prepare-e2e.sh`, this is why `hack/prepare-e2e.sh` is executed before `e2e run -c test/e2e/e2e.yaml`. For other tools, I think we can add them in the `steps` section.

```
       - quay.io/jetstack/cert-manager-cainjector:v1.3.1
       - quay.io/jetstack/cert-manager-controller:v1.3.1
       - quay.io/jetstack/cert-manager-webhook:v1.3.1
       - docker.io/kubesphere/kube-rbac-proxy:v0.4.1
       - docker.io/apache/skywalking-oap-server:8.8.1
       - docker.io/apache/skywalking-ui:8.8.1
       - ghcr.io/apache/skywalking-swck-spring-demo:v0.0.1
       - docker.io/apache/skywalking-java-agent:8.8.0-java8
```

These are all public images, right?

For public Docker images, they can be pulled inside KinD when needed, so you don't have to pull firstly

Reply:
Yes, these are public images. But I have tested the `e2e` in the local machine and in the github action. In the local machine, we need to pull the image and load the images to kind, then the `e2e` can work, If we don't do so, the `e2e` may not work. In the github action, it can work whether we load the images to kind. I don’t know if my environment settings cause the `e2e` to fail.

Reply:
This looks like some DNS or network issue about docker pull :) I used to face one on local KinD due to accidentally set up proxy for the terminal of running KinD.

Reply:
>  I don’t know if my environment settings cause the e2e to fail.

I tend to believe that that's your local env issue. Let's ensure e2e is working, then I will help you fix that.

Reply:
OK, thanks a lot.